cpp/include/copc-lib/io/laz_reader.hpp

```diff
@@ -0,0 +1,59 @@
+#ifndef COPCLIB_IO_LAZ_READER_H_
+#define COPCLIB_IO_LAZ_READER_H_
+
+#include "copc-lib/io/base_reader.hpp"
+
+namespace copc::laz
```

Comment:
Looks like we use copc::las for the config stuff and copc::laz for the reader/writer haha pretty confusing, then the config is `LazConfig`, but then the header is `LasHeader`, I wonder if we should just put it all as laz/Laz? Less stuff to remember :smile: